src/jats/importer/jats-comments.ts

```diff
@@ -50,175 +37,60 @@ export const isJATSComment = (node: Node) => {
 export const parseJATSComment = (node: Node): JATSComment | undefined => {
   const text = node.textContent
   if (text) {
-    const id = /id="(.+)"/.exec(text)
     const queryText = /queryText="(.+)"/.exec(text)
-    if (id && queryText) {
+    if (queryText) {
+      const parentNode = node.parentNode as Element
+      const index = parentNode.outerHTML.indexOf(queryText[1])
       return {
-        id: id[1],
+        id: generateID(ObjectTypes.CommentAnnotation),
         text: queryText[1],
+        index,
       }
     }
   }
 }
 
-/**
- * Replaces processing instructions with tokens
- */
-export const markComments = (doc: Document): JATSCommentMark[] => {
-  const marks: JATSCommentMark[] = []
+export const markComments = (doc: Document) => {
   const root = doc.getRootNode()
   const queue: Node[] = [root]
+  const commentsElement = doc.createElement('comments-annotations')
   while (queue.length !== 0) {
     const node = queue.shift()
     if (node) {
       if (isJATSComment(node)) {
         const comment = parseJATSComment(node)
         if (comment) {
-          const token = addMark(doc, node)
-          if (token) {
-            const mark = {
-              token,
-              comment,
-            }
-            marks.push(mark)
-          }
+          const highlightMarker = createHighlightMarkerElement(doc, comment.id)
+          node.parentNode?.insertBefore(highlightMarker, node)
+          const commentElement = createCommentElement(doc, comment)
+          commentsElement.appendChild(commentElement)
         }
       }
       node.childNodes.forEach((child) => {
         queue.push(child)
       })
     }
   }
-  return marks
-}
 
-const addMark = (doc: Document, node: Node) => {
-  const parent = node.parentElement
-  if (parent) {
-    const token = uuidv4()
-    const tokenNode = doc.createTextNode(token)
-    parent.insertBefore(tokenNode, node)
-    return token
+  if (commentsElement.hasChildNodes()) {
+    doc.documentElement.appendChild(commentsElement)
   }
 }
 
-export const createComments = (
-  models: Model[],
-  marks: JATSCommentMark[]
-): CommentAnnotation[] => {
-  const comments = []
-  for (const model of models) {
-    if (isHighlightableModel(model)) {
-      comments.push(...processModel(model, marks))
-    } else if (isKeyword(model)) {
-      comments.push(...processKeyword(model, marks))
-    }
-  }
-  return comments
+const createHighlightMarkerElement = (doc: Document, id: string) => {
+  const highlightMarker = doc.createElement('highlight-marker')
+  highlightMarker.setAttribute('id', id)
+  highlightMarker.setAttribute('position', 'point')
```

Comment:
Debugging this i only found that we use `point` comments on the import. Please correct me if I'm mistaken